### PR TITLE
Phonics table of contents

### PR DESCRIPTION
**Pull request by willsaid**
Your Letters View Controller used to display Letters A-Z, but this now displays each individual Phonic (Sound), along with color coding, new thumbnails, and some other small things.

### CODE REVIEW

**Comment by calda** [comment]
Looking good!

Some comments:
 - When I opened up the project, I got this error: <br>screenshot 2018-05-26 at 8 18 29 am <br> You added a reference to that file without actually moving the file into the project folder. 

 - I think this green color should be a tad darker (it's pretty tough to read) <br> screenshot 2018-05-26 at 8 21 58 am

- The `ProgressBar` feature on the `LetterCell`s in `LetterViewController` isn't set up right anymore. It's still showing progress for the letter that the sound is associated with. That needs to be updated to show the progress of the individual sound.

Other than those small issues, I think it's pretty good! Thanks for working on this.

File: Phonics/Content Model/PHContent.swift [hunk 1]
```diff
@@ -111,7 +116,26 @@ class PHContentManager {
...
                 
-                let sound = Sound(sourceLetter: letter,
+                //parse color for each Sound
+                let colorString = line[7].trimmingWhitespace()
+                var color: UIColor
+                switch colorString {
+                case "orange":
+                    color = .orange
+                case "green":
+                    color = .green
```
Comment (calda) [inline Phonics/Content Model/PHContent.swift]: This is the one to change. `.green` is a bit too bright. Probably will need to be a custom shade of green

File: Phonics/Content Model/PHContent.swift [hunk 2]
```diff
@@ -181,6 +207,11 @@ class PHContentManager {
         })
     }
     
+    //phonics sorted according to CSV and chart ordering
+    func allPhonicsSorted() -> [Sound] {
```
Comment (calda) [inline Phonics/Content Model/PHContent.swift]: Change this to a dynamic variable (e.g. `var allPhonicsSorted [Sound]: { ... }`)

File: Phonics/View Controllers/LettersViewController.swift [hunk 3]
```diff
@@ -18,6 +19,8 @@ class LettersViewController: UIViewController, UICollectionViewDataSource, UICol
     static func present(from source: UIViewController, with difficulty: Letter.Difficulty) {
         let controller = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: storyboardId) as! LettersViewController
         controller.difficulty = difficulty
+        controller.phonics = PHContent.allPhonicsSorted()
```
Comment (calda) [inline Phonics/View Controllers/LettersViewController.swift]: `PHContent` is a singleton, so `PHContent.allPhonicsSorted` is always available in any scope. No need for a `controller.phonics` instance variable.

File: Phonics/View Controllers/LettersViewController.swift [hunk 4]
```diff
@@ -36,12 +43,21 @@ class LettersViewController: UIViewController, UICollectionViewDataSource, UICol
     //MARK: - Collection View Data Source
     
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return PHLetters.count
+        if difficulty == .easyDifficulty {
+            return PHLetters.count
+        }
+        
+        return phonics.count
```
Comment (calda) [inline Phonics/View Controllers/LettersViewController.swift]: Just access it through the singleton instead. `PHContent.allPhonicsSorted.count`

File: Phonics/View Controllers/LettersViewController.swift [hunk 5]
```diff
@@ -36,12 +43,21 @@ class LettersViewController: UIViewController, UICollectionViewDataSource, UICol
...
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "letter", for: indexPath) as! LetterCell
-        cell.decorateForLetter(PHLetters[indexPath.item], difficulty: difficulty)
+        if self.difficulty == .easyDifficulty {
+            cell.decorateForLetter(PHLetters[indexPath.item], difficulty: difficulty)
+        } else {
+            let phonic = phonics[indexPath.item]
```
Comment (calda) [inline Phonics/View Controllers/LettersViewController.swift]: Just access it through the singleton instead. `PHContent.allPhonicsSorted[indexPath.item]`

**Comment by willsaid** [comment]
Fixed up small stuff from the PR feedback

**Review by calda — APPROVED** [review]
Looks great, thanks!

**Pull request merged**